code/cranking/update-pricing/update-pricing.preview.en.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * This calls zeta's permissionless update pricing function through the Exchange object.
+ * Cranks zeta's on-chain pricing ensuring all our greeks and theos are up-to-date.
+ */
+async function updatePricing() {
+  // Get relevant expiry indices.
+  for (var asset of assetList) {
+    let indicesToCrank = [];
+    if (!Exchange.getZetaGroup(asset).perpsOnly) {
```

Comment:
this has all been simplified quite a bit recently - see the example here: https://github.com/zetamarkets/sdk/blob/main/examples/cranking/cranking-example.ts#L147

you no longer need to worry about expiryseries, because thats a legacy options thing. just call `Exchange.updatePricing` on each asset

Reply:
done